src/modules/rover_ackermann/AckermannPosControl/AckermannPosControl.cpp

```diff
@@ -0,0 +1,390 @@
+		_rover_attitude_setpoint_pub.publish(rover_attitude_setpoint);
+	}
+}
+
+void AckermannPosControl::updateAutoSubscriptions()
+{
+	if (_home_position_sub.updated()) {
+		home_position_s home_position{};
+		_home_position_sub.copy(&home_position);
+		_home_position = Vector2d(home_position.lat, home_position.lon);
+	}
+
+	if (_position_setpoint_triplet_sub.updated()) {
+		updateWaypointsAndAcceptanceRadius();
+	}
+
+	if (_mission_result_sub.updated()) {
+		mission_result_s mission_result{};
+		_mission_result_sub.copy(&mission_result);
+		_mission_finished = mission_result.finished;
+	}
+}
+
+void AckermannPosControl::updateWaypointsAndAcceptanceRadius()
+{
+	position_setpoint_triplet_s position_setpoint_triplet{};
+	_position_setpoint_triplet_sub.copy(&position_setpoint_triplet);
+
+	RoverControl::globalToLocalSetpointTriplet(_curr_wp_ned, _prev_wp_ned, _next_wp_ned, position_setpoint_triplet,
+			_curr_pos_ned,
+			_home_position, _global_ned_proj_ref);
+
+	_prev_waypoint_transition_angle = _waypoint_transition_angle;
+	_waypoint_transition_angle = RoverControl::calcWaypointTransitionAngle(_prev_wp_ned, _curr_wp_ned, _next_wp_ned);
+
+	// Update acceptance radius
+	_prev_acceptance_radius = _acceptance_radius;
+
+	if (_param_ra_acc_rad_max.get() >= _param_nav_acc_rad.get()) {
+		_acceptance_radius = updateAcceptanceRadius(_waypoint_transition_angle, _param_nav_acc_rad.get(),
+				     _param_ra_acc_rad_gain.get(), _param_ra_acc_rad_max.get(), _param_ra_wheel_base.get(), _param_ra_max_str_ang.get());
+
+	} else {
+		_acceptance_radius = _param_nav_acc_rad.get();
+	}
+
+	// Waypoint cruising speed
+	_cruising_speed = position_setpoint_triplet.current.cruising_speed > 0.f ? math::constrain(
+				  position_setpoint_triplet.current.cruising_speed, 0.f, _param_ro_max_speed.get()) : _param_ro_max_speed.get();
+}
+
+float AckermannPosControl::updateAcceptanceRadius(const float waypoint_transition_angle,
+		const float default_acceptance_radius, const float acceptance_radius_gain,
+		const float acceptance_radius_max, const float wheel_base, const float max_steer_angle)
+{
+	// Calculate acceptance radius s.t. the rover cuts the corner tangential to the current and next line segment
+	float acceptance_radius = default_acceptance_radius;
+	const float theta = waypoint_transition_angle / 2.f;
+	const float min_turning_radius = wheel_base / sinf(max_steer_angle);
+	const float acceptance_radius_temp = min_turning_radius / tanf(theta);
+	const float acceptance_radius_temp_scaled = acceptance_radius_gain *
+			acceptance_radius_temp; // Scale geometric ideal acceptance radius to account for kinematic and dynamic effects
+	acceptance_radius = math::constrain<float>(acceptance_radius_temp_scaled, default_acceptance_radius,
+			    acceptance_radius_max);
+
+	// Publish updated acceptance radius
+	position_controller_status_s pos_ctrl_status{};
+	pos_ctrl_status.acceptance_radius = acceptance_radius;
+	pos_ctrl_status.timestamp = hrt_absolute_time();
```

Comment:
```suggestion
	pos_ctrl_status.timestamp = _timestamp;
```
To be consistent with the other publications

Reply:
Applied in [454995a](https://github.com/PX4/PX4-Autopilot/pull/24285/commits/454995a3a15f362273c6bc3da6fc9edb9e088165)